Add clarification about AWS Connection behaviour (#29341)
mistercrunch/Airflow: docs/apache-airflow-providers-amazon/connections/aws.rst

Amazon Web Services Connection
==============================

The Amazon Web Services connection type enables the :ref:`AWS Integrations <AWS>`.

> .. important:: Amazon Web Services Connection could be tested in the UI/API or by call

Reviewer: ```suggestion
.. important:: The Amazon Web Services Connection can be tested in the UI/API or by calling
```

    :meth:`~airflow.providers.amazon.aws.hooks.base_aws.AwsGenericHook.test_connection`,

>     it is **important** to correct interpret result of this test.

Reviewer: ```suggestion
    it is **important** to correctly interpret the result of this test.
```

    During this test components of Amazon Provider invoke AWS Security Token Service API
    `GetCallerIdentity <https://docs.aws.amazon.com/STS/latest/APIReference/API_GetCallerIdentity.html>`__.

>     This service **only** could check is your credentials valid or not.

Reviewer 2: ```suggestion
    This service can  **only** check if your credentials are valid.
```

>     Unfortunately it is not possible to validate is this credentials has access to specific AWS service or not.

Reviewer 2: ```suggestion
    Unfortunately it is not possible to validate if credentials have access to specific AWS service or not.
```

>     If you use Amazon Provider for communicate with AWS API compatible services (MinIO, LocalStack, etc.)

Reviewer: ```suggestion
    If you use the Amazon Provider to communicate with AWS API compatible services (MinIO, LocalStack, etc.)
```

    test connection failure **doesn't mean** that your connection has wrong credentials.

>     Many of compatible services provide only limited amount of AWS API services,
>     and most of them not implement AWS STS GetCallerIdentity method.

Reviewer: ```suggestion
    Many compatible services provide only a limited number of AWS API services,
    and most of them do not implement the AWS STS GetCallerIdentity method.
```

To use IAM instance profile, create an "empty" connection (i.e. one with no AWS Access Key ID or AWS Secret Access Key
specified, or ``aws://``).

Default Connection IDs
-----------------------

The default connection ID is ``aws_default``. If the environment/machine where you are running Airflow has the
file credentials in ``/home/.aws/``, and the default connection has user and pass fields empty, it will take
automatically the credentials from there.

.. important:: Previously, the ``aws_default`` connection had the "extras" field set to ``{"region_name": "us-east-1"}``
    on install. This means that by default the ``aws_default`` connection used the ``us-east-1`` region.
    This is no longer the case and the region needs to be set manually, either in the connection screens in Airflow,
    or via the ``AWS_DEFAULT_REGION`` environment variable.

.. caution:: If you do not set ``[database] load_default_connections`` to ``True``

>     most probably you do not have ``aws_default``. By historical reason Amazon Provider

Reviewer: ```suggestion
    most probably you do not have ``aws_default``. For historical reasons, the Amazon Provider
```

>     components (Hooks, Operators, Sensors, etc.) fallback to default boto3 credentials strategy
>     in case of missing Connection ID, this behaviour is deprecated and will be removed in a future releases.

Reviewer: ```suggestion
    components (Hooks, Operators, Sensors, etc.) fallback to the default boto3 credentials strategy
    in case of a missing Connection ID. This behaviour is deprecated and will be removed in a future releases.
```

>     If you need use default boto3 credential strategy (credentials in environment variables, IAM Profile, etc.)
>     please provide ``None`` instead of connection id.

Reviewer: ```suggestion
    If you need to use the default boto3 credential strategy (credentials in environment variables, IAM Profile, etc.)
    please provide ``None`` instead of a connection ID.
```
